addons/hr_work_entry/models/hr_version.py

```diff
@@ -449,7 +449,7 @@ def write(self, vals):
         dependent_fields = self._get_fields_that_recompute_we()
         salary_simulation = self.env.context.get('salary_simulation')
         if not salary_simulation and any(key in dependent_fields for key in vals):
-            for version in self:
+            for version in self.sudo():
```

Comment:
same here for naming as we return stuff
```suggestion
            for version_sudo in self.sudo():
```